WordPress/src/main/java/org/wordpress/android/viewmodel/themes/ThemesViewModel.kt

```diff
@@ -0,0 +1,59 @@
+package org.wordpress.android.viewmodel.themes
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import kotlinx.coroutines.CoroutineDispatcher
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.model.ThemeModel
+import org.wordpress.android.fluxc.store.ThemeStore
+import org.wordpress.android.modules.UI_THREAD
+import org.wordpress.android.util.AppLog
+import org.wordpress.android.util.AppLog.T.THEMES
+import org.wordpress.android.viewmodel.Event
+import org.wordpress.android.viewmodel.ScopedViewModel
+import org.wordpress.android.viewmodel.themes.ThemesViewModel.Selection.KEEP_CURRENT_HOMEPAGE
+import javax.inject.Inject
+import javax.inject.Named
+
+class ThemesViewModel @Inject constructor(
+    private val themeStore: ThemeStore,
+    @Named(UI_THREAD) private val mainDispatcher: CoroutineDispatcher
+) : ScopedViewModel(mainDispatcher) {
+    enum class Selection {
+        USE_THEME_HOMEPAGE,
+        KEEP_CURRENT_HOMEPAGE
+    }
+
+    var bottomSheetSelection = KEEP_CURRENT_HOMEPAGE
+    var site = SiteModel()
```

Comment:
We make the two above variables private and expose less of the internals of our view model.
Another approach would be to create a State for our ViewModel. That way it would be easier to write tests that verify that each action moves the state to an expected value.

Reply:
Thank you for tackling this 👍